frontend/syntax.asdl

```diff
@@ -263,9 +263,9 @@ module syntax
 
   # Each arm tests one word against multiple words
   CaseArm = (List[word] pat_list, List[command] action, List[int] spids)
-  IfArm = (condition cond, List[command] action, List[int] spids)
+  IfArm = (Token keyword, condition cond, Token? then_kw, List[command] action, List[int] spids)
```

Comment:
How about adding comments that

- keyword is either `if` or `elif` 
- `then` is optional for YSH / Oil ?